### ISSUE TITLE
Match start of method name

### ISSUE THREAD

**Issue by tjohn** [yuki24/did_you_mean#50]
We had an issue where no matches were returned despite there being a string match.

```ruby
class Foo
  def self.set_result_count; end
end

Foo.set_result
#=> NoMethodError: undefined method `set_result' for Foo:Class

Foo.set_result_c
#=> NoMethodError: undefined method `set_result_c' for Foo:Class

    Did you mean? .set_result_count
```

It would be desirable for the suggestion to appear if everything matches except the postfix. I am not sure how this would look on shorter method names.

**Comment by yuki24** [comment]
@tjohn Actually it was already implemented (more specifically [Jaro-Winkler distance](http://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance)), but it often suggests too many suggestions (see #49). I guess the combination of Levenshtein and Jaro-Winkler distance would be a good fit.

It's hard to make did_you_mean smarter, but the next major version should be coming with a better suggestion algorithm.

Thanks!

**Comment by yuki24** [comment]
I'm closing this issue since a prefix-based spellchecker has been implemented and shipped with [v0.10.0](https://github.com/yuki24/did_you_mean/releases/tag/v0.10.0). Thanks!
